discovery-provider/src/utils/redis_cache.py

```diff
@@ -19,23 +19,31 @@ def extract_key(path, arg_items):
     key = f"{cache_prefix}:{path}:{req_args}"
     return key
 
-def use_redis_cache(key, ttl_sec, work_func):
-    """Attemps to return value by key, otherwise caches and returns `work_func`"""
-    redis = redis_connection.get_redis()
+def get_pickled_key(redis, key):
```

Comment:
NIT - key is a bit misleading. maybe `get_pickled_response`?

Reply:
Will rename